source/shared/core_stmt.cpp

```diff
@@ -796,6 +798,9 @@ bool core_sqlsrv_fetch( _Inout_ sqlsrv_stmt* stmt, _In_ SQLSMALLINT fetch_orient
         // move to the record requested.  For absolute records, we use a 0 based offset, so +1 since 
         // SQLFetchScroll uses a 1 based offset, otherwise for relative, just use the fetch_offset provided.
         SQLRETURN r = stmt->current_results->fetch( fetch_orientation, ( fetch_orientation == SQL_FETCH_RELATIVE ) ? fetch_offset : fetch_offset + 1 TSRMLS_CC );
+        if ( stmt->cursor_type == SQL_CURSOR_FORWARD_ONLY && stmt->conn->ce_option.enabled == true ) {
+            stmt->fwd_row_index++;
+        }
```

Comment:
It's not necessary to reset fwd_row_index explicitly. past_fetch_end is set to true so the driver can raise an error when the user tries to fetch again even after SQL_NO_DATA. past_fetch_end and fwd_row_index is reset when the statement is re-executed and a new result set is created (https://github.com/v-kaywon/msphpsql/blob/22b4817f658a16edb8d24eba7eba2b2e44e2b01c/source/shared/core_stmt.cpp#L217)